subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -275,4 +284,29 @@ public void buildCache(Action<? super BuildCacheConfiguration> action) {
     public BuildCacheConfigurationInternal getBuildCache() {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public void pluginRepositories(Action<? super PluginRepositoriesSpec> pluginSettings) {
+        pluginSettings.execute(getPluginRepositoriesSpec());
+    }
+
+    @Override
+    public void pluginManagement(Action<? super PluginManagementSpec> rule) {
+        rule.execute(getPluginManagementSpec());
+    }
+
+    @Override
+    public PluginRepositoriesSpec getPluginRepositoriesSpec() {
+        Instantiator instantiator = services.get(Instantiator.class);
```

Comment:
I tried that, but couldn't get a FileResolver that had a base dir so you could do relative paths. Do you know how to get a base dir to make this work? I tried StartParameters but rootDir was null.